Mathlib/Data/Multiset/DershowitzManna.lean

```diff
@@ -0,0 +1,292 @@
+/-
+Copyright (c) 2024 Haitian Wang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Haitian Wang, Malvin Gattinger
+-/
+import Mathlib.Algebra.Order.Sub.Basic
+import Mathlib.Data.Multiset.OrderedMonoid
+
+/-!
+# Dershowitz-Manna ordering
+
+In this file we define the _Dershowitz-Manna ordering_ on multisets. Specifically, for two multisets
+`M` and `N` in a partial order `(S, <)`, `M` is smaller than `N` in the Dershowitz-Manna ordering if
+`M` can be obtained from `N` by replacing one or more elements in `N` by some finite number of
+elements from `S`, each of which is smaller (in the underling ordering over `S`) than one of the
+replaced elements from `N`. We prove that, given a well-founded partial order on the underlying set,
+the Dershowitz-Manna ordering defined over multisets is also well-founded.
+
+## Main results
+
+- `Multiset.IsDershowitzMannaLT` : the standard definition fo the `Dershowitz-Manna ordering`.
+- `Multiset.wellFounded_isDershowitzMannaLT` : the main theorem about the
+`Dershowitz-Manna ordering` being well-founded.
+- `Multiset.TransLT_eq_isDershowitzMannaLT` : two definitions of the `Dershowitz-Manna ordering`
+are equivalent.
+
+## References
+
+* [Wikipedia, Dershowitz–Manna ordering*]
+(https://en.wikipedia.org/wiki/Dershowitz%E2%80%93Manna_ordering)
+
+* [CoLoR](https://github.com/fblanqui/color), a Coq library on rewriting theory and termination.
+  Our code here is inspired by their formalization and the theorem is called `mOrd_wf` in the file
+  [MultisetList.v](https://github.com/fblanqui/color/blob/1.8.5/Util/Multiset/MultisetOrder.v).
+
+-/
+
+open Relation
+
+namespace Multiset
+
+variable {α : Type*} [Preorder α] {M N P : Multiset α} {a : α}
+
+/-- The standard Dershowitz–Manna ordering. -/
+def IsDershowitzMannaLT (M N : Multiset α) : Prop :=
+  ∃ X Y Z,
+      Z ≠ ∅
+    ∧ M = X + Y
+    ∧ N = X + Z
+    ∧ ∀ y ∈ Y, ∃ z ∈ Z, y < z
+
+/-- `IsDershowitzMannaLT` is transitive. -/
+lemma IsDershowitzMannaLT.trans :
+    IsDershowitzMannaLT M N → IsDershowitzMannaLT N P → IsDershowitzMannaLT M P := by
+  classical
+  rintro ⟨X₁, Y₁, Z₁, hZ₁, rfl, rfl, hYZ₁⟩ ⟨X₂, Y₂, Z₂, hZ₂, hXZXY, rfl, hYZ₂⟩
```

Comment:
The `hZ₁` here is unused and can be replaced with `_`. For some reason when copying the file into another project (where I use a slightly older Lean version) the linter warned about this, but did not do so here in CI :thinking:.

Reply:
You can even replace it with `-` to make sure it's not even introduced